modules/lang-painless/src/main/java/org/elasticsearch/painless/MethodWriter.java

```diff
@@ -262,6 +262,50 @@ public void writeToStrings() {
         }
     }
 
+    /** Writes a dynamic binary instruction: returnType, lhs, and rhs can be different */
+    public void writeDynamicBinaryInstruction(Location location, Type returnType, Type lhs, Type rhs, Operation operation) {
+        org.objectweb.asm.Type descriptor = org.objectweb.asm.Type.getMethodType(returnType.type, lhs.type, rhs.type);
```

Comment:
I would change the name here to somthing like methodType.
In addition: This looks like messy code duplication. Maybe just use a simple wrapper method for the invokeDynamic, to not copy paste each line?